internal/dependencytrack/project.go

```diff
@@ -31,9 +32,35 @@ type ProjectMetrics struct {
 	InheritedRiskScore float64 `json:"inheritedRiskScore"`
 }
 
-// GetProjects returns a list of all projects
-func (c *Client) GetProjects() ([]*Project, error) {
-	req, err := c.newRequest(http.MethodGet, "/api/v1/project", nil)
+// GetAllProjects returns a list of all projects with the help of pagination
+func (c *Client) GetAllProjects() ([]*Project, error) {
+	// dependency track per default only returns a 100 items per get, therefore we need to iterate over allProjects pages to get allProjects projects
+
+	// allProjects found in pagination
+	allProjects := []*Project{}
+	// the last project found in the last pagination page result
+	lastPaginationPage := 1
+	// state var to show if allProjects projects where found
+	foundAll := false
+
+	for !foundAll {
+		req, err := c.GetProjects(lastPaginationPage, 100)
+		if err != nil {
+			return nil, err
+		}
+		if len(req) == 0 {
```

Comment:
The number of total items will not necessarily be a multiple of 100, which means the 'last page' will have less than 100 results. 

If we change this check to `len(req) < 100` (and move some of the logic around) I think we can avoid an extra request to the API in most cases.